### PR TITLE
Feat/store form app state

### PR DESCRIPTION
**Pull request by bjmccotter7192**
Main things trying to get solved:
- handleUpdate from context (typing errors currently but kinda works)
- Write Test that validates the formData fields are being added when a new Web Component is added to the Form
- Anything else you think looks off/bad/wrong

### CODE REVIEW

File: examples/form-state/index.jsx [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  Router,
+  Page,
+  FormContext
+} from '@department-of-veterans-affairs/va-forms-system-core';
+
+import '@department-of-veterans-affairs/component-library/dist/main.css';
+import { defineCustomElements } from '@department-of-veterans-affairs/component-library';
+
+// eslint-disable-next-line @typescript-eslint/no-unsafe-call
+void defineCustomElements();
```
Comment (micahchiang) [inline examples/form-state/index.jsx:16]: 👀 what does this do? Not sure I understand.

**Review by micahchiang — COMMENTED** [review]
It's looking good, nice work @bjmccotter7192!

Could you drop a link to the zenhub ticket in the description for this PR, as well as a short summary of what context is and why we need it for this library? Thanks. 

I think we might also need to think a bit more about how we should approach the `optionalUpdateHandler`. I know we put it in so we could test, but on further thinking that feels sort of like testing for the sake of testing. We should ask ourselves - Why would our end users want to provide their own handler? What use cases would it cover? If we can better answer those, we may be able to better handle the optional updater.

File: src/routing/Router.tsx [hunk 2]
```diff
@@ -1,16 +1,35 @@
...
 export default function Router(props: RouterProps): JSX.Element {
+  const [formData, handleUpdate] = useState({});
+
+  const updateFormData = (data: Record<string, unknown>) => {
+    if (props.optionalHandleUpdate) {
+      handleUpdate(props.optionalHandleUpdate);
+    } else {
+      const updatedData = { ...formData, ...data };
+      handleUpdate(updatedData);
+    }
```
[the remark this answers is not in the document]
Comment (bjmccotter7192) [reply src/routing/Router.tsx:20]: I'll remove this for now and if we find a need for it later we can add it.

File: src/routing/FormContext.tsx [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import { createContext } from 'react';
+import { IFormContextType } from './types';
+
+export const defaultState = {
+  formData: {
+    foo: 'Testing',
```
[the remark this answers is not in the document]
Comment (bjmccotter7192) [reply src/routing/FormContext.tsx:6]: Changing this to `formData: {}` and removing Tests for FormContext.

**Review by micahchiang — APPROVED** [review]
LGTM, thanks @bjmccotter7192!

**Pull request merged**